llmfoundry/callbacks/kill_loss_spike_callback.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Track training runs for loss spikes or persistently high training loss."""
+from __future__ import annotations
+
+import logging
+from collections import deque
+
+import numpy as np
+import torch
+from composer.core import Callback, State, TimeUnit
+from composer.loggers import Logger, MosaicMLLogger
+from composer.utils import dist
+
+from llmfoundry.utils.exceptions import HighLossError, LossSpikeError
+from llmfoundry.utils.warnings import experimental_class
+
+log = logging.getLogger(__name__)
+
+__all__ = ['KillLossSpike']
+
+MIN_WINDOW_SIZE = 100
```

Comment:
might as well make all of these configurable in the callback with defaults so that if you want to try different values its a yaml change not a code change

Reply:
Sure

Reply:
Made loss_cap and window_size configurable